gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/StructuredQuery.java

```diff
@@ -714,32 +687,34 @@ B addGroupBy(String property, String... others) {
       return self();
     }
 
-    B mergeFrom(DatastoreV1.Query queryPb) {
+    B mergeFrom(com.google.datastore.v1beta3.Query queryPb) {
       if (queryPb.getKindCount() > 0) {
         kind(queryPb.getKind(0).getName());
       }
-      if (queryPb.hasStartCursor()) {
+      if (!queryPb.getStartCursor().isEmpty()) {
         startCursor(new Cursor(queryPb.getStartCursor()));
       }
-      if (queryPb.hasEndCursor()) {
+      if (!queryPb.getEndCursor().isEmpty()) {
         endCursor(new Cursor(queryPb.getEndCursor()));
       }
-      if (queryPb.hasOffset()) {
-        offset(queryPb.getOffset());
-      }
+      offset(queryPb.getOffset());
       if (queryPb.hasLimit()) {
-        limit(queryPb.getLimit());
+        limit(queryPb.getLimit().getValue());
       }
       if (queryPb.hasFilter()) {
-        filter(Filter.fromPb(queryPb.getFilter()));
+        Filter currFilter = Filter.fromPb(queryPb.getFilter());
+        if (currFilter != null) {
+          filter(currFilter);
+        }
       }
-      for (DatastoreV1.PropertyOrder orderByPb : queryPb.getOrderList()) {
+      for (com.google.datastore.v1beta3.PropertyOrder orderByPb : queryPb.getOrderList()) {
         addOrderBy(OrderBy.fromPb(orderByPb));
       }
-      for (DatastoreV1.PropertyExpression projectionPb : queryPb.getProjectionList()) {
+      for (com.google.datastore.v1beta3.Projection projectionPb
+           : queryPb.getProjectionList()) {
         addProjection(Projection.fromPb(projectionPb));
       }
-      for (DatastoreV1.PropertyReference groupByPb : queryPb.getGroupByList()) {
+      for (com.google.datastore.v1beta3.PropertyReference groupByPb : queryPb.getDistinctOnList()) {
```

Comment:
should we keep calling it Group By?

@pcostell  

Is there no plan to re-add aggregation functions?

Does a user must specify distinct on all projection value or is the an implicit "first" aggregation?